src/test/java/uk/gov/companieshouse/exemptions/service/ExemptionsApiServiceTest.java

```diff
@@ -61,63 +56,77 @@ class ExemptionsApiServiceTest {
     @Mock
     private ChangedResource changedResource;
 
-    @InjectMocks
     private ExemptionsApiService exemptionsApiService;
 
+    @BeforeEach
+    void setup() {
+        exemptionsApiService = new ExemptionsApiService(apiClientSupplier, logger, mapper);
+    }
+
     @Test
     @DisplayName("Test should successfully invoke chs-kafka-api")
     void invokeChsKafkaApi() throws ApiErrorResponseException {
-        when(apiClientService.getInternalApiClient()).thenReturn(internalApiClient);
+        // given
+        when(apiClientSupplier.get()).thenReturn(internalApiClient);
         when(internalApiClient.privateChangedResourceHandler()).thenReturn(privateChangedResourceHandler);
         when(privateChangedResourceHandler.postChangedResource(any(), any())).thenReturn(changedResourcePost);
         when(changedResourcePost.execute()).thenReturn(response);
         when(mapper.mapChangedResource(resourceChangedRequest)).thenReturn(changedResource);
 
-        ServiceStatus serviceStatus = exemptionsApiService.invokeChsKafkaApi(resourceChangedRequest);
+        // when
+        exemptionsApiService.invokeChsKafkaApi(resourceChangedRequest);
 
-        assertEquals(ServiceStatus.SUCCESS, serviceStatus);
-
-        verify(apiClientService).getInternalApiClient();
+        // then
+        verify(apiClientSupplier).get();
         verify(internalApiClient).privateChangedResourceHandler();
         verify(privateChangedResourceHandler).postChangedResource("/private/resource-changed", changedResource);
         verify(changedResourcePost).execute();
     }
 
     @Test
-    @DisplayName("Test should handle a service unavailable exception when response code is HTTP 503")
+    @DisplayName("Throw service unavailable exception when response code is HTTP 503")
```

Comment:
Line 86 to 126 can be incorporated into one parameterised test like this (with a better name for the test):

```
    @ParameterizedTest
    @CsvSource({
            "503 , Service Unavailable",
            "500 , Internal Service Error",
            "200 , ''"
    })
    void test(final int statusCode, final String statusMessage) throws ApiErrorResponseException {
        // given
        setupExceptionScenario(statusCode, statusMessage);

        // when
        Executable actual = () -> exemptionsApiService.invokeChsKafkaApi(resourceChangedRequest);

        // then
        assertThrows(ServiceUnavailableException.class, actual);
        verifyExceptionScenario();
    }
```